src/components/common/OfficeAddresses.vue

```diff
@@ -4,51 +4,35 @@
-            :editing="false"
-          />
-          <div v-else>(Not entered)</div>
+          <div v-if="isEmptyAddress(mailingAddress)">(Not entered)</div>
+          <MailingAddress v-else :address="mailingAddress" :editing="false" />
         </v-col>
 
-        <v-col md="4">
+        <v-col md="4" class="pr-4">
           <label class="delivery-address-header"><strong>Delivery Address</strong></label>
-          <delivery-address
-            v-if="!isEmptyAddress(deliveryAddress) && !inheritMailingAddress"
-            :address="deliveryAddress"
-            :editing="false"
-          />
-          <div v-else-if="isEmptyAddress(deliveryAddress)">(Not entered)</div>
-          <div v-else>Same as Mailing Address</div>
+          <div v-if="isEmptyAddress(deliveryAddress)">(Not entered)</div>
+          <div v-else-if="isSame(mailingAddress, deliveryAddress)">Same as Mailing Address</div>
+          <DeliveryAddress v-else :address="deliveryAddress" :editing="false" />
         </v-col>
       </v-row>
 
       <v-row no-gutters id="summary-records-address" v-if="!entityFilter(CorpTypeCd.COOP)" class="mt-4">
         <v-col md="3" class="mr-n1"><label><strong>Records Office</strong></label></v-col>
-        <v-col md="4">
+        <v-col md="4" class="pr-4">
           <label class="mailing-address-header"><strong>Mailing Address</strong></label>
-          <mailing-address
-            v-if="!inheritRegisteredAddress && !isEmptyAddress(recMailingAddress)"
-            :address="recMailingAddress"
-            :editing="false"
-          />
-          <div v-else-if="isEmptyAddress(recMailingAddress)">(Not entered)</div>
-          <div v-else>Same as Registered Office</div>
+          <div v-if="isEmptyAddress(recMailingAddress)">(Not entered)</div>
+          <div v-else-if="isSame(mailingAddress, recMailingAddress)">Same as Registered Office</div>
+          <MailingAddress v-else :address="recMailingAddress" :editing="false" />
         </v-col>
 
-        <v-col md="4">
+        <v-col md="4" class="pr-4">
           <label class="delivery-address-header"><strong>Delivery Address</strong></label>
-          <delivery-address
-            v-if="!inheritRecMailingAddress && !inheritRegisteredAddress && !isEmptyAddress(recDeliveryAddress)"
-            :address="recDeliveryAddress"
-            :editing="false"
-          />
-          <div v-else-if="isEmptyAddress(recDeliveryAddress)">(Not entered)</div>
-          <div v-else-if="inheritRegisteredAddress">Same as Registered Office</div>
-          <div v-else>Same as Mailing Address</div>
+          <div v-if="isEmptyAddress(recDeliveryAddress)">(Not entered)</div>
+          <div v-else-if="isSame(deliveryAddress, recDeliveryAddress)">Same as Registered Office</div>
+          <div v-else-if="isSame(recMailingAddress, recDeliveryAddress)">Same as Mailing Address</div>
+          <DeliveryAddress v-else :address="recDeliveryAddress" :editing="false" />
```

Comment:
^^^ simplified logic!

I didn't do this frivolously... I was already in this area fixing a bug where the inherit flags were not being updated (since the component on this page is another instance and doesn't have the same class variables as the editable address component). So instead I check if the addresses are the same (which is also what happens if the user types in the same address multiple times then saves/reloads the draft: the checkbox is automatically checked).